Vizzuality/half-earth-v3
# Feature/caching strategies

This PR creates a custom Service Worker. 
The service worker gets added to the `bundled` code after the `build` process. This step is triggered by the `build-service-worker` script and the config of the worker is defined on the `service-worker-build.js` file. That generates a `custom-service-worker.js` file that follows the structure set in place on the `service-worker-template.js`.
This last file is the one that should be modified in order to update the service worker. It has a version number that should be updated on every modification.
The files cached on `installing` phase ([service worker lifecycle for reference](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle)) currently are the app own `js`, `css`, `svg` and `html` assets.
Once the service worker is active it will start intercepting the browser requests and it will currently be listening to cache images and fonts requests (from the app url). So those assets will not be served from the cache until a reload of the page have taken place.
Eventually some other requests could be intercepted to increase the assets cached.

Have tried to cache ArcGIS Js API requests but the worker in not working as expected.

I've added a new dependency: `workbox-build`

To test it locally run: 
`yarn build`
then open the build folder on a server (I'm using `serve`), do not forget to open it on `localhost:3000` as is one of the urls whitelisted on the Biodiversity Facets layer.
`serve build -l tcp://localhost:3000`
and check the `Network` and `Application` tabs on your browser console (you should see the Service worker there and the cache :)

![image](https://user-images.githubusercontent.com/6906348/63854691-022c8a80-c99e-11e9-9c61-2089737ec805.png)
